packages/grpc-web/index.d.ts

```diff
@@ -32,31 +32,71 @@ declare module "grpc-web" {
   }
 
   export class ClientReadableStream<Response> {
-    on (type: "error",
+    on (eventType: "error",
         callback: (err: Error) => void): ClientReadableStream<Response>;
-    on (type: "status",
+    on (eventType: "status",
         callback: (status: Status) => void): ClientReadableStream<Response>;
-    on (type: "metadata",
+    on (eventType: "metadata",
         callback: (status: Metadata) => void): ClientReadableStream<Response>;
-    on (type: "data",
+    on (eventType: "data",
         callback: (response: Response) => void): ClientReadableStream<Response>;
-    on (type: "end",
+    on (eventType: "end",
         callback: () => void): ClientReadableStream<Response>;
+    on (eventType: string,
+        callback: any): ClientReadableStream<Response>;
 
-    removeListener (type: "error",
+    removeListener (eventType: "error",
                     callback: (err: Error) => void): void;
-    removeListener (type: "status",
+    removeListener (eventType: "status",
                     callback: (status: Status) => void): void;
-    removeListener (type: "metadata",
+    removeListener (eventType: "metadata",
                     callback: (status: Metadata) => void): void;
-    removeListener (type: "data",
+    removeListener (eventType: "data",
                     callback: (response: Response) => void): void;
-    removeListener (type: "end",
+    removeListener (eventType: "end",
                     callback: () => void): void;
+    removeListener (eventType: string,
+                    callback: any): void;
                     
     cancel (): void;
   }
 
+  export interface StreamInterceptor<Req, Resp> {
+    intercept(request: Request<Req, Resp>,
+              invoker: (request: Request<Req, Resp>) =>
+      ClientReadableStream<Resp>): ClientReadableStream<Resp>;
+  }
+  

```

Comment:
Thanks for putting this up! I was wondering if you could also add UnaryInterceptor while you're at it.

From: https://github.com/grpc/grpc-web/blob/master/javascript/net/grpc/web/interceptor.js#L47

```suggestion

  export interface UnaryInterceptor<Req, Resp> {
      intercept(request: Request<Req, Resp>,
                invoker: (request: Request<Req, Resp>) =>
        Promise<!UnaryResponse<Resp>>): Promise<UnaryResponse<Resp>>;
    }
  
```

Reply:
Yes literally working on that part as we speak :)

Reply:
@willsalz Added the `UnaryInterceptor` interface typings to the PR. Still WIP. Hopefully should land soon.